test/acceptance/visibility-test.js

```diff
@@ -0,0 +1,124 @@
+import {
+  acceptance,
+  exists,
+  updateCurrentUser,
+} from "discourse/tests/helpers/qunit-helpers";
+import { test } from "qunit";
+import { click, visit } from "@ember/test-helpers";
+
+acceptance("Welcome Link Banner - Logged out", function () {
+  test("banner can be hidden from anons", async function (assert) {
+    settings.hide_for_anon = true;
+    await visit("/latest");
+
+    assert.ok(
+      !exists(".welcome-link-banner-wrapper"),
+      "hides the banner for anons"
+    );
+  });
+
+  test("banner can be shown to anons", async function (assert) {
+    settings.hide_for_anon = false;
+    await visit("/latest");
+
+    assert.ok(
+      exists(".welcome-link-banner-wrapper"),
+      "hides the banner for anons"
+    );
```

Comment:
…and in the inverse cases:

```suggestion
    assert
      .dom(".welcome-link-banner-wrapper")
      .exists("hides the banner for anons");
```